cmd/ocsp-updater/main.go

```diff
@@ -177,34 +187,54 @@ func (updater *OCSPUpdater) generateResponse(status core.CertificateStatus) (res
 
 	ocspResponse, err := updater.cac.GenerateOCSP(signRequest)
 	if err != nil {
-		return responseMeta{}, err
+		return core.CertificateStatus{}, err
 	}
 
-	timestamp := updater.clk.Now()
-	status.OCSPLastUpdated = timestamp
-	ocspResp := &core.OCSPResponse{
-		Serial:    cert.Serial,
-		CreatedAt: timestamp,
-		Response:  ocspResponse,
-	}
-	return responseMeta{ocspResp, &status}, nil
+	status.OCSPLastUpdated = updater.clk.Now()
+	status.OCSPResponse = ocspResponse
+	return status, nil
 }
 
-func (updater *OCSPUpdater) storeResponse(tx *gorp.Transaction, meta responseMeta) error {
-	// Record the response.
-	err := tx.Insert(meta.OCSPResponse)
+func (updater *OCSPUpdater) generateRevokedResponse(status core.CertificateStatus) (core.CertificateStatus, error) {
+	cert, err := updater.sac.GetCertificate(status.Serial)
 	if err != nil {
-		return err
+		return core.CertificateStatus{}, err
 	}
 
-	// Reset the update clock
-	_, err = tx.Update(meta.CertificateStatus)
+	signRequest := core.OCSPSigningRequest{
+		CertDER:   cert.DER,
+		Status:    string(core.OCSPStatusRevoked),
+		Reason:    status.RevokedReason,
+		RevokedAt: status.RevokedDate,
+	}
+
+	ocspResponse, err := updater.cac.GenerateOCSP(signRequest)
 	if err != nil {
-		return err
+		return core.CertificateStatus{}, err
 	}
 
-	// Done
-	return nil
+	now := updater.clk.Now()
+	status.OCSPLastUpdated = now
+	status.OCSPResponse = ocspResponse
+	return status, nil
+}
+
+func (updater *OCSPUpdater) storeResponse(status core.CertificateStatus, statusGuard core.OCSPStatus) error {
+	// Update the certificateStatus table with the new OCSP response, the status
+	// WHERE is used make sure we don't overwrite a revoked response with a one
+	// containing a 'good' status and that we don't do the inverse when the OCSP
+	// status should be 'good'.
+	_, err := updater.dbMap.Exec(
```

Comment:
Have you tested to see what happens when the `WHERE` is not satisfied? Do we get an error, or does it silently ignore the update? I think we want the latter behavior.